File: pharmaB2B-frontend/pharmab2b/src/screens/contact-us/api/contactUs.js
import axios from 'axios';

const contactUs = async (name, email, question, phoneNumber) => {
  const payload = {
    name: name,
    email: email,
    question: question,
    contact_no: phoneNumber ? phoneNumber : '',
  };
  return await axios.post('api/customers/contact_us/', payload);
};

export default contactUs;

File: pharmaB2B-backend/pharmaB2B/customers/models/contact_us.py
from django.db import models
from phonenumber_field.modelfields import PhoneNumberField

from pharmaB2B.core.models import Base


class ContactUs(Base):

    name = models.TextField()

    contact_no = PhoneNumberField(null=True, region="GB", blank=True)

    question = models.TextField()

    email = models.EmailField()

    responded = models.BooleanField(default=False)

    class Meta:

        verbose_name = "Contact Us"

        verbose_name_plural = "Contacted Us"
